auth/couchdb_session_authenticator.go

```diff
@@ -155,49 +160,52 @@ func (a *CouchDbSessionAuthenticator) Authenticate(request *http.Request) error
 // getCookie returns an AuthSession cookie to be used in a request.
 // A new cookie will be fetched from the session end-point when needed.
 func (a *CouchDbSessionAuthenticator) getCookie() (*http.Cookie, error) {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	a.flushRefreshChannel()
+
 	if a.session == nil || !a.session.isValid() {
-		err := a.syncRequestSession()
+		newSession, err := a.requestSession()
 		if err != nil {
 			return nil, err
 		}
+		a.session = newSession
 	} else if a.session.needsRefresh() {
-		ch := make(chan error)
+		// start a background process to refresh the session.
+		// the refreshed session will be passed to a buffered channel
+		// and updated in a next request at flushRefreshChannel() call.
 		go func() {
-			ch <- a.requestSession()
-		}()
-		select {
-		case err := <-ch:
+			// we are intentionally not returning errors to the parent process
+			// to avoid raisng error to a client with still valid session.
```

Comment:
Nit:
```suggestion
			// to avoid raising error to a client with still valid session.
```

Reply:
Fixed in https://github.com/IBM/cloudant-go-sdk/pull/74/commits/246d485ac866f7db9df54519841ecf5400650c10